cmake/SociConfig.cmake

```diff
@@ -69,7 +69,7 @@ else()
             set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
         elseif (CMAKE_SYSTEM_NAME MATCHES "Windows")
             # We need to disable the warnings for using %I64d with MinGW.
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-pedantic-ms-format")
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-pedantic-ms-format -Wno-format")
```

Comment:
@jsonn You're right. But I've also checked MinGW64 v4.8.3 (i686 - 32 bit) it still fails with error

```
[ 78%] Building CXX object src/backends/odbc/CMakeFiles/soci_odbc.dir/standard-i
nto-type.cpp.obj
cd /d C:\WORK\GITHUB\soci\b.mgw32\src\backends\odbc && C:\MinGW\mingw-w64\i686-4
.8.3-posix-sjlj-rt_v3-rev2\mingw32\bin\g++.exe   -DBOOST_ALL_NO_LIB -DDBUG_OFF -
DHAVE_BOOST=1 -DHAVE_ODBC=1 -DSOCI_ABI_VERSION=\"4_0\" -DSOCI_DLL -pedantic -Wer
ror -Wall -Wpointer-arith -Wcast-align -Wcast-qual -Wfloat-equal -Wredundant-dec
ls -Wno-long-long -std=gnu++98  -Wno-pedantic-ms-format @CMakeFiles/soci_odbc.di
r/includes_CXX.rsp   -o CMakeFiles\soci_odbc.dir\standard-into-type.cpp.obj -c C
:\WORK\GITHUB\soci\src\backends\odbc\standard-into-type.cpp
C:\WORK\GITHUB\soci\src\backends\odbc\standard-into-type.cpp: In member function
 'virtual void soci::odbc_standard_into_type_backend::post_fetch(bool, bool, soc
i::indicator*)':
C:\WORK\GITHUB\soci\src\backends\odbc\standard-into-type.cpp:175:53: error: ISO
C++98 does not support the 'I' scanf flag [-Werror=format=]
           if (sscanf(buf_, "%" LL_FMT_FLAGS "d", &ll) != 1)
                                                     ^
C:\WORK\GITHUB\soci\src\backends\odbc\standard-into-type.cpp:175:53: error: form
at '%d' expects argument of type 'int*', but argument 3 has type 'long long int*
' [-Werror=format=]
C:\WORK\GITHUB\soci\src\backends\odbc\standard-into-type.cpp:183:53: error: ISO
C++98 does not support the 'I' scanf flag [-Werror=format=]
           if (sscanf(buf_, "%" LL_FMT_FLAGS "u", &ll) != 1)
                                                     ^
C:\WORK\GITHUB\soci\src\backends\odbc\standard-into-type.cpp:183:53: error: form
at '%u' expects argument of type 'unsigned int*', but argument 3 has type 'long
long unsigned int*' [-Werror=format=]
cc1plus.exe: all warnings being treated as errors
```

So, I agree. We'd better move from sprintf/printf to some C++ code instead of playing with ifdefs.